pages/oauth/authorize.js

```diff
@@ -15,12 +15,12 @@ import SignInOrJoinFree from '../../components/SignInOrJoinFree';
 import { useLoggedInUser } from '../../components/UserProvider';
 
 const applicationQuery = gqlV2`
-  query OAuthAuthorization($clientId: String!) {
+  query OAuthAuthorization($clientId: String!, $withoutLoggedInUser: Boolean!) {
     application(clientId: $clientId) {
       id
       name
-      clientId
-      redirectUri
+      clientId @skip(if: $withoutLoggedInUser)
+      redirectUri @skip(if: $withoutLoggedInUser)
```

Comment:
What's happening here?

How is the next developer seeing this gonna understand the rationale?

Reply:
Sorry I should have explained this. The API will not allow to return `clientId` and `redirectUri` if the user is not logged in. So we skip them in that case. I've added a comment to better explain this.

Reply:
That should not be the case, please modify the API and remove this code.

Reply:
By the way, you're seeing these when you're logged in, but that's because you're the creator of the application, that would not work for a normal user.